### PR TITLE
Add Cost Explorer API findings

### PR DESCRIPTION
**Pull request by brendasukh**
## Hey, I just made a Pull Request!
- Adding some docs on how you can use the AWS Cost Explorer API to implement a client for the Cost Insights plugin.
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4698-s8tpyfqr7oyj6uvn3kxkoeumnxupuwtr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037bd6afcb22d1eee604e56

**Review by adamdmharvey — COMMENTED** [review]
This is a fantastic contrib, thanks @brendasukh !!  👏  I can definitely see using this if/when we get here.

I might wonder if adding a minor appendix on the main README pointing out the existance of the contrib folder might at least elevate it's awareness a bit more, and then leave users to fish around for things they may find helpful in there.

Thanks again!

File: plugins/cost-insights/contrib/aws-cost-explorer-api.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
+# Using the AWS Cost Explorer API with Cost Insights
+
+Cost Insights implementers are currently expected to write their own Cost Insights client to interact with the frontend plugin. We've looked into short-term solutions which we've since punted on, but wanted to share our findings from our exploration of the AWS Cost Explorer API.
+
+**Note:** Each request using the Cost Explorer API will incur a cost of \$0.01.
+`
```
Comment (dhenneke) [inline plugins/cost-insights/contrib/aws-cost-explorer-api.md:6]: 👀 
```suggestion
```

File: plugins/cost-insights/contrib/aws-cost-explorer-api.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
+# Using the AWS Cost Explorer API with Cost Insights
+
+Cost Insights implementers are currently expected to write their own Cost Insights client to interact with the frontend plugin. We've looked into short-term solutions which we've since punted on, but wanted to share our findings from our exploration of the AWS Cost Explorer API.
```
Comment (timbonicus) [inline plugins/cost-insights/contrib/aws-cost-explorer-api.md:3]: Hmm, `punted` isn't the most rosy term 😅  What about something like:

> Cost Insights currently does not provide a CostInsightsApi client out of the box, this is left up to the implementor. We plan to provide an open-source client and backend eventually that can pull cost data from different cloud providers. In the meantime, we briefly explored the AWS Cost Explorer API and wanted to share our findings. Please contribute to this documentation if you try any experiments with AWS Cost Explorer.

File: plugins/cost-insights/contrib/aws-cost-explorer-api.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+**Note:** Each request using the Cost Explorer API will incur a cost of \$0.01.
```
Comment (timbonicus) [inline plugins/cost-insights/contrib/aws-cost-explorer-api.md:5]: Maybe add something like:

> If you anticipate high usage, adding some caching of Cost Explorer API responses could prove worthwhile over time.

File: plugins/cost-insights/contrib/aws-cost-explorer-api.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+      ]
+    }
+  ]
+}
+
+```
+
+## Setup
+
+1. Install the AWS Cost Explorer SDK. The AWS docs recommend using the SDK over making calls to the API directly as it simplifies authentication and provides direct access to commands.
```
Comment (timbonicus) [inline plugins/cost-insights/contrib/aws-cost-explorer-api.md:34]: I'd say avoid numbering if there's just one thing in the section.

File: plugins/cost-insights/contrib/aws-cost-explorer-api.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+
+There are a couple of options for breaking down costs for a resource.
+
+1. We can use `USAGE_TYPE` (units used to measure usage for each service) supported by the `GroupBy` field. This requires the lowest amount of effort as we can easily add it to the existing resource request. Examples usage types for EC2: BoxUsage:c1.medium(Hrs), BoxUsage:m3.xlarge(Hrs), and BoxUsage:t1.micro(Hrs)
+
+2. Similar to resources, we can use custom tags to group the data. For example, you’re able to tag objects within a bucket so that would be one way to get more granularity within a resource. More info on object tagging [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/object-tagging.html).
+
+## Cost Allocation Tags
+
+Tags are labels that you can assign to an AWS resource, each with a key and value. You can use cost allocation tags to track your AWS costs on a detailed level. You can tag resources through the AWS UI by using the Tags tab on the relevant resource screen. You can also add tags progrmatically via the relevant cloud product API.
```
Comment (timbonicus) [inline plugins/cost-insights/contrib/aws-cost-explorer-api.md:300]:
```suggestion
Tags are labels that you can assign to an AWS resource, each with a key and value. You can use cost allocation tags to track your AWS costs on a detailed level. You can tag resources through the AWS UI by using the Tags tab on the relevant resource screen. You can also add tags programatically via the relevant cloud product API.
```

**Review by timbonicus — APPROVED** [review]
Awesome write-up, @brendasukh ! 🔥

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4698-s8tpyfqr7oyj6uvn3kxkoeumnxupuwtr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037bd6afcb22d1eee604e56

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4698-s8tpyfqr7oyj6uvn3kxkoeumnxupuwtr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037bd6afcb22d1eee604e56

File: .github/styles/vocab.txt [hunk 6]
```diff
@@ -215,6 +216,7 @@ prebaked
 preconfigured
 prepack
 productional
+programatically
```
Comment (Rugvip) [inline .github/styles/vocab.txt:219]: Not sure about this one https://en.m.wiktionary.org/wiki/programatically 😅

File: .github/styles/vocab.txt [hunk 7]
```diff
@@ -207,6 +207,7 @@ onboarding
 pagerduty
 pageview
 parallelization
+params
```
Comment (Rugvip) [inline .github/styles/vocab.txt:210]: Prolly wrap this in a code block to avoid having to add it here, otherwise spell out parameters imo

Comment (brendasukh) [reply .github/styles/vocab.txt:210]: @Rugvip sounds good, updated

File: .github/styles/vocab.txt (hunk 6 above)
Comment (brendasukh) [reply .github/styles/vocab.txt:219]: 😅oof

**Comment by brendasukh** [comment]
@adamdmharvey thanks for the feedback! Added a note in the main Cost Insights README :)

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4698-s8tpyfqr7oyj6uvn3kxkoeumnxupuwtr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037bd6afcb22d1eee604e56

File: .github/styles/vocab.txt [hunk 8]
```diff
@@ -276,6 +278,7 @@ validators
 varchar
 winston
 www
+xlarge
```
Comment (Rugvip) [inline .github/styles/vocab.txt:281]: Would say do the same with this btw, `BoxUsage:m3.xlarge(Hrs)` instead of adding to vocab

**Review by Rugvip — COMMENTED** [review]
Clean up vocab and merge? :grin:

**Review by Rugvip — APPROVED** [review]

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4698-s8tpyfqr7oyj6uvn3kxkoeumnxupuwtr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6037bd6afcb22d1eee604e56

**Pull request merged**